src/rules/delimiterDangle.js

```diff
@@ -0,0 +1,95 @@
+import _ from 'lodash';
+
+export default (context) => {
+  const option = context.options[0] || 'never';
+  const sourceCode = context.getSourceCode();
+
+  const reporter = (node, message, fix) => {
+    return () => {
+      context.report({
+        fix,
+        message,
+        node
+      });
+    };
+  };
+
+  const makeReporters = (node, tokenToFix) => {
+    return {
+      dangle: reporter(node, 'Unexpected trailing delimiter', (fixer) => {
+        return fixer.replaceText(tokenToFix, '');
+      }),
+      noDangle: reporter(node, 'Missing trailing delimiter', (fixer) => {
+        return fixer.insertTextAfter(tokenToFix, ',');
+      })
+    };
+  };
+
+  const evaluate = (node, lastChildNode) => {
+    if (!lastChildNode) {
+      return;
+    }
+
+    const [penultimateToken, lastToken] = sourceCode.getLastTokens(node, 2);
+
+    const isDangling = [';', ','].indexOf(penultimateToken.value) > -1;
+    const isMultiLine = penultimateToken.loc.start.line !== lastToken.loc.start.line;
+
+    const report = makeReporters(lastChildNode, penultimateToken);
+
+    if (option === 'always' && !isDangling) {
+      report.noDangle();
```

Comment:
To avoid unnecessary if condition checks, each successful if condition should return, i.e.,

```js
if (option === 'always' && !isDangling) {
	report.noDangle();

	return;
}
```